### PR TITLE
Intel Benchmark P2P support

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Added IMB-P2P test case in RDMA/IB test suite

### [Test result]
**<IBM>**
[LISAv2 Test Results Summary]
Test Run On           : 04/20/2019 20:25:43
ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:2:12

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-IBM-MPI-2VM                                                            PASS                128.7 
	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-1-FirstBoot : IMB-P2P : PASS** 
	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-2-Reboot : IMB-P2P : PASS** 
	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 

Logs can be found at C:\LISAv2\RZ98\TestResults\2019-20-04-20-25-36-8707

04/20/2019 22:37:55 : [INFO ] Creating 'C:\LISAv2\RZ98\Azure-RZ98-TestLogs.zip' from 'C:\LISAv2\RZ98\TestResults\2019-20-04-20-25-36-8707'
04/20/2019 22:37:55 : [INFO ] C:\LISAv2\RZ98\Azure-RZ98-TestLogs.zip created successfully.
04/20/2019 22:37:55 : [INFO ] Analyzing results..
04/20/2019 22:37:55 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2.
04/20/2019 22:37:59 : [INFO ] Cleaning up C:\LISAv2\RZ98
04/20/2019 22:37:59 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2
04/20/2019 22:37:59 : [INFO ] LISAv2 exit code: 0

**<Intel>**
[LISAv2 Test Results Summary]
Test Run On           : 04/20/2019 20:25:43
ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:3:53

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-INTEL-MPI-2VM                                                          PASS               230.28 
	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-1-FirstBoot : IMB-P2P : PASS** 
	InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-2-Reboot : IMB-P2P : PASS** 
	InfiniBand-Verification-2-Reboot : IMB-RMA : PASS 
	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 

Logs can be found at C:\LISAv2\ZV71\TestResults\2019-20-04-20-25-37-4067

04/21/2019 00:19:41 : [INFO ] Creating 'C:\LISAv2\ZV71\Azure-ZV71-TestLogs.zip' from 'C:\LISAv2\ZV71\TestResults\2019-20-04-20-25-37-4067'
04/21/2019 00:19:42 : [INFO ] C:\LISAv2\ZV71\Azure-ZV71-TestLogs.zip created successfully.
04/21/2019 00:19:42 : [INFO ] Analyzing results..
04/21/2019 00:19:42 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@3.
04/21/2019 00:19:45 : [INFO ] Cleaning up C:\LISAv2\ZV71
04/21/2019 00:19:45 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@3
04/21/2019 00:19:45 : [INFO ] LISAv2 exit code: 0

**<MVAPICH>**
[LISAv2 Test Results Summary]
Test Run On           : 04/20/2019 20:25:43
ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:41

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-MVAPICH-MPI-2VM                                                        PASS                37.67 
	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-1-FirstBoot : IMB-P2P : PASS** 
	InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
	**InfiniBand-Verification-2-Reboot : IMB-P2P : PASS** 
	InfiniBand-Verification-2-Reboot : IMB-RMA : PASS 
	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 

Logs can be found at C:\LISAv2\PJ19\TestResults\2019-20-04-20-25-37-4817

04/20/2019 21:07:04 : [INFO ] Creating 'C:\LISAv2\PJ19\Azure-PJ19-TestLogs.zip' from 'C:\LISAv2\PJ19\TestResults\2019-20-04-20-25-37-4817'
04/20/2019 21:07:04 : [INFO ] C:\LISAv2\PJ19\Azure-PJ19-TestLogs.zip created successfully.
04/20/2019 21:07:04 : [INFO ] Analyzing results..
04/20/2019 21:07:04 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@4.
04/20/2019 21:07:07 : [INFO ] Cleaning up C:\LISAv2\PJ19
04/20/2019 21:07:07 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@4
04/20/2019 21:07:07 : [INFO ] LISAv2 exit code: 0

<Open>

### CODE REVIEW

File: Testscripts/Linux/TestRDMA_MultiVM.sh [hunk 1]
```diff
@@ -270,6 +271,69 @@ function Run_IMB_NBC() {
...
+					LogMsg "$mpi_run_path -hostlist $modified_slaves:$VM_Size -np $(($VM_Size * $total_virtual_machines)) -e MPI_IB_PKEY=$MPI_IB_PKEY $imb_p2p_path $extra_params"
+					$mpi_run_path -hostlist $modified_slaves:$VM_Size -np $(($VM_Size * $total_virtual_machines)) -e MPI_IB_PKEY=$MPI_IB_PKEY $imb_p2p_path $extra_params > IMB-P2P-AllNodes-output-Attempt-${attempt}.txt
+				;;
+				open)
+					LogMsg "$mpi_run_path --allow-run-as-root --host $master,$slaves -n $(($p2p_ppn * $total_virtual_machines)) $mpi_settings $imb_p2p_path $extra_params"
+					$mpi_run_path --allow-run-as-root --host $master,$slaves -n $(($p2p_ppn * $total_virtual_machines)) $mpi_settings $imb_p2p_path $extra_params > IMB-P2P-AllNodes-output-Attempt-${attempt}.txt
+				;;
+				intel)
+					LogMsg "$mpi_run_path -hosts $master,$modified_slaves -ppn $p2p_ppn -n $(($VM_Size * $total_virtual_machines)) $mpi_settings $imb_p2p_path $extra_params"
+					$mpi_run_path -hosts $master,$modified_slaves -ppn $p2p_ppn -n $(($VM_Size * $total_virtual_machines)) $mpi_settings $imb_p2p_path $extra_params > IMB-P2P-AllNodes-output-Attempt-${attempt}.txt
```
Comment (adriansuhov) [inline Testscripts/Linux/TestRDMA_MultiVM.sh]: $modified_slaves variable is only available in IBM test cases, not Intel. Should change to $slaves

**Review by adriansuhov — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/TestRDMA_MultiVM.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/TestRDMA_MultiVM.sh]: fixed in 67eadb55a6cd62eb271249df2decdecad6729662

**Comment by juhlee-microsoft** [comment]
Verified Intel MPI with the latest change.

[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] [LISAv2 Test Results Summary]
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] Test Run On           : 04/22/2019 19:02:45
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] Total Time (dd:hh:mm) : 0:3:32
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] -------------------------------------------------------------------------------------------------------------------------------------------
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus]     1 INFINIBAND           INFINIBAND-INTEL-MPI-2VM                                                          PASS               208.32 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-P2P : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-P2P : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-RMA : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] Logs can be found at C:\LISAv2\GO80\TestResults\2019-22-04-19-02-41-5234
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:53 : [INFO ] Creating 'C:\LISAv2\GO80\Azure-GO80-TestLogs.zip' from 'C:\LISAv2\GO80\TestResults\2019-22-04-19-02-41-5234'
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:53 : [INFO ] C:\LISAv2\GO80\Azure-GO80-TestLogs.zip created successfully.
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:53 : [INFO ] Analyzing results..
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:53 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2.
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:56 : [INFO ] Cleaning up C:\LISAv2\GO80
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:56 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@2
[Azure INFINIBAND-INTEL-MPI-2VM southcentralus] 04/22/2019 22:34:56 : [INFO ] LISAv2 exit code: 0

**Comment by ader1990** [comment]
@juhlee-microsoft why are there 2 PRs with almost same content? Or you want to merge this as the first and the second one will have lesser commits?

**Review by adriansuhov — APPROVED** [review]

**Pull request merged**